src/helpers/__tests__/get-insights-anonymous-user-token-test.ts

```diff
@@ -0,0 +1,37 @@
+import getInsightsAnonymousUserToken, {
+  ANONYMOUS_TOKEN_COOKIE_KEY,
+} from '../get-insights-anonymous-user-token';
+
+const DAY = 86400000; /* 1 day in ms*/
+const DATE_TOMORROW = new Date(Date.now() + DAY).toUTCString();
+const DATE_YESTERDAY = new Date(Date.now() - DAY).toUTCString();
+
+const resetCookie = cookieKey => {
+  document.cookie = `${cookieKey}=;expires=Thu, 01-Jan-1970 00:00:01 GMT;`;
+};
+
+describe('getInsightsAnonymousUserToken', () => {
+  beforeEach(() => {
+    resetCookie(ANONYMOUS_TOKEN_COOKIE_KEY);
+  });
+
+  it('should return undefined when no cookies', () => {
+    expect(getInsightsAnonymousUserToken()).toBe(undefined);
+  });
+
+  it('should return undefined when cookie present but expired', () => {
+    document.cookie = `${ANONYMOUS_TOKEN_COOKIE_KEY}=anonymous-uuid;expires=${DATE_YESTERDAY};`;
+    expect(getInsightsAnonymousUserToken()).toBe(undefined);
+  });
+
+  it('should return the anonymous uuid when cookie present and valid', () => {
+    document.cookie = `${ANONYMOUS_TOKEN_COOKIE_KEY}=anonymous-uuid;expires=${DATE_TOMORROW};`;
+    expect(getInsightsAnonymousUserToken()).toBe('anonymous-uuid');
+  });
+
+  it("should not care about other cookies and fail if they're malformed", () => {
+    document.cookie = `${ANONYMOUS_TOKEN_COOKIE_KEY}=anonymous-uuid;expires=${DATE_TOMORROW};`;
```

Comment:
I think this line is not intended.

Reply:
is it intended. 
All cookies "live" in the same string, which needs to be parsed.
The spirit of this test is to check `getCookie(cookieName_1)` will not fail if `cookieName_2` is malformed: 
- it should not care about other cookie 
- it should not fail if other cookies are malformed.
Should the description of the test be improved?

Reply:
https://github.com/algolia/instantsearch.js/pull/4279/files#r364202991